### PR TITLE
feat: add automated actions tables

### PR DESCRIPTION
**Pull request by gastonfournier**
Add tables to support automated actions based on the [design document](https://docs.google.com/document/d/15xPgn6B8gzMPk9RjMAUGu8XvozQ_45jjrsVl_Bm6Chg/edit?usp=sharinghttps://docs.google.com/document/d/15xPgn6B8gzMPk9RjMAUGu8XvozQ_45jjrsVl_Bm6Chg/edit?usp=sharing)

### CODE REVIEW

File: src/migrations/20240111125100-automated-actions.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        (
+            id SERIAL PRIMARY KEY NOT NULL,
+            created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+            created_by_user_id INTEGER,
+            name varchar(255),
+            actor_id INTEGER,
+            source varchar(255),
+            source_id INTEGER,
+            payload JSONB DEFAULT '{}'::jsonb NOT NULL,
+            FOREIGN KEY (project) references projects(id) ON DELETE CASCADE,
```
Comment (nunogois) [inline src/migrations/20240111125100-automated-actions.js:14]: `project` needs to be added as a column in the table before defining it as a foreign key.

File: src/migrations/20240111125100-automated-actions.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        CREATE TABLE IF NOT EXISTS actions
+        (
+            id SERIAL PRIMARY KEY NOT NULL,
+            created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+            created_by_user_id INTEGER,
+            action_set_id INTEGER references action_sets (id) ON DELETE CASCADE,
+            sort_order INTEGER,
+            action varchar(255) NOT NULL,
+            execution_params JSONB DEFAULT '{}'::jsonb NOT NULL,
```
Comment (nunogois) [inline src/migrations/20240111125100-automated-actions.js:25]: Another trailing comma ☝️

File: src/migrations/20240111125100-automated-actions.js (hunk 1 above)
Comment (gastonfournier) [reply src/migrations/20240111125100-automated-actions.js:14]: Yes! :D

**Review by nunogois — APPROVED** [review]
LGTM!

**Pull request merged**